src/muya/lib/contentState/arrowCtrl.js

```diff
@@ -48,6 +48,18 @@ const arrowCtrl = ContentState => {
     return null
   }
 
+  ContentState.prototype.getTable = function (cell) {
```

Comment:
I think this method below is what you want, and can replace `getTable`.

```javascript
  // tableBlockCtrl.js
  ContentState.prototype.getTableBlock = function () {
    const { start, end } = this.cursor
    const startBlock = this.getBlock(start.key)
    const endBlock = this.getBlock(end.key)
    const startParents = this.getParents(startBlock)
    const endParents = this.getParents(endBlock)
    const affiliation = startParents
      .filter(p => endParents.includes(p))

    if (affiliation.length) {
      const table = affiliation.find(p => p.type === 'figure')
      return table
    }
  }
```

and type `figure` block is the parent of type `table` block. you should always use `figure` block(has a table block) as actual `table`.

Reply:
you are 100% right, thanks. replaced